src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -1487,6 +1486,11 @@ async ValueTask<T> GetFieldValueSequential<T>(int column, bool async, Cancellati
             }
         }
 
+        internal Task<T> GetFieldValue<T>(string name, bool async, CancellationToken cancellationToken = default)
```

Comment:
This (and IsDBNull below) are basically unnecessary... Asynchronous column access is only needed for very large rows, where CommandBehavior.SequentialAccess is specified when executing the command. Without SequentialAccess, Npgsql will always buffer (at least) the entire row, meaning that any column access is always synchronous. So you can safely call the synchronous GetFieldValue and IsDBNull even from the sync path.